pkg/controller/util/overrides.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+
+	"github.com/kubernetes-sigs/federation-v2/pkg/apis/core/typeconfig"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type ClusterOverrides struct {
+	Path      []string
+	Overrides map[string]interface{}
+}
+
+func NewClusterOverrides(typeConfig typeconfig.Interface, override *unstructured.Unstructured) (*ClusterOverrides, error) {
+	overrides, path, err := marshallOverrides(typeConfig, override)
+	if err != nil {
+		return nil, err
+	}
+	return &ClusterOverrides{
+		Path:      path,
+		Overrides: overrides,
+	}, nil
+}
+
+func marshallOverrides(typeConfig typeconfig.Interface, override *unstructured.Unstructured) (map[string]interface{}, []string, error) {
```

Comment:
(no action required) is marshall the right term here? It seems like we're deserializing the unstructured data into a meaningful map and string slice.